internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go

```diff
@@ -0,0 +1,631 @@
+				state.SourceLocation = metadata.ResourceData.Get("source_location").(string)
+				state.PointInTimeInUTC = metadata.ResourceData.Get("point_in_time_in_utc").(string)
+				state.CoordinatorPublicIPAccessEnabled = *props.CoordinatorEnablePublicIPAccess
+				state.CoordinatorServerEdition = *props.CoordinatorServerEdition
+				state.CoordinatorStorageQuotaInMb = *props.CoordinatorStorageQuotaInMb
+				state.CoordinatorVCoreCount = *props.CoordinatorVCores
+				state.HaEnabled = *props.EnableHa
+				state.NodeCount = *props.NodeCount
+				state.NodePublicIPAccessEnabled = *props.NodeEnablePublicIPAccess
+				state.NodeServerEdition = *props.NodeServerEdition
+				state.NodeStorageQuotaInMb = *props.NodeStorageQuotaInMb
+				state.NodeVCores = *props.NodeVCores
+				state.ShardsOnCoordinatorEnabled = *props.EnableShardsOnCoordinator
+
+				if v := props.CitusVersion; v != nil {
+					state.CitusVersion = *v
+				}
+
+				if v := props.MaintenanceWindow; v != nil {
+					state.MaintenanceWindow = flattenMaintenanceWindow(v)
+				}
+
+				if v := props.PreferredPrimaryZone; v != nil {
+					state.PreferredPrimaryZone = *v
+				}
+
+				if v := props.PostgresqlVersion; v != nil {
+					state.SqlVersion = *v
+				}
+
+				if v := props.EarliestRestoreTime; v != nil {
+					state.EarliestRestoreTime = *v
+				}
+			}
+
+			if model.Tags != nil {
+				state.Tags = *model.Tags
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r CosmosDbPostgreSQLClusterResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 3 * time.Hour,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cosmos.ClustersClient
+
+			id, err := clusters.ParseServerGroupsv2ID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandMaintenanceWindow(input []MaintenanceWindow) *clusters.MaintenanceWindow {
+	if len(input) == 0 {
+		return &clusters.MaintenanceWindow{
+			CustomWindow: utils.String("Disabled"),
+		}
+	}
+
+	v := input[0]
+
+	maintenanceWindow := clusters.MaintenanceWindow{
+		CustomWindow: utils.String("Enabled"),
+		StartHour:    utils.Int64(v.StartHour),
+		StartMinute:  utils.Int64(v.StartMinute),
+		DayOfWeek:    utils.Int64(v.DayOfWeek),
+	}
+
+	return &maintenanceWindow
+}
+
+func flattenMaintenanceWindow(input *clusters.MaintenanceWindow) []MaintenanceWindow {
+	if input == nil || input.CustomWindow == nil || *input.CustomWindow == "Disabled" {
+		return nil
+	}
+
+	result := MaintenanceWindow{}
+
+	if input.DayOfWeek != nil {
+		result.DayOfWeek = *input.DayOfWeek
+	}
+
+	if input.StartHour != nil {
+		result.StartHour = *input.StartHour
+	}
+
+	if input.StartMinute != nil {
+		result.StartMinute = *input.StartMinute
+	}
+
+	return []MaintenanceWindow{result}
```

Comment:
And those `pointer.From` can be thrown in here.
```suggestion
	return []MaintenanceWindow{MaintenanceWindow{
	  StartMinute: pointer.From(input.StartMinute)
	  .
	  .
	  .
	}}
```

Reply:
Updated